s3kor.go

```diff
@@ -151,28 +151,23 @@ func setUpLogger() {
 }
 
 func getAwsConfig() aws.Config {
-	if *pCustomEndpointUrl == "" {
-		config := aws.Config{
-			CredentialsChainVerboseErrors: aws.Bool(true),
-			MaxRetries:                    aws.Int(30),
-		}
+	config := aws.Config{
+		CredentialsChainVerboseErrors: aws.Bool(true),
+		MaxRetries:                    aws.Int(30),
+	}
 
-		if *pRegion != "" {
+	if pCustomEndpointURL == nil || (pCustomEndpointURL != nil && *pCustomEndpointURL == "") {
+		if pRegion != nil && *pRegion != "" {
 			config.Region = aws.String(*pRegion)
 		}
-
 		return config
 	}
 
-	fmt.Printf("Using custom endpoint [%+v] on region [%+v]\n", *pCustomEndpointUrl, *pRegion)
-	config := aws.Config{
-		Endpoint:                      aws.String(*pCustomEndpointUrl),
-		CredentialsChainVerboseErrors: aws.Bool(true),
-		MaxRetries:                    aws.Int(30),
-		S3ForcePathStyle:              aws.Bool(true),
-	}
+	fmt.Printf("Using custom endpoint [%+v] on region [%+v]\n", *pCustomEndpointURL, *pRegion)
+	config.Endpoint = aws.String(*pCustomEndpointURL)
+	config.S3ForcePathStyle = aws.Bool(true)
 
-	if *pRegion != "" {
+	if pRegion != nil && *pRegion != "" {
```

Comment:
this is just a simplification of the logic, and some `!= nil` checks to ensure it doesn't panic